Packet++/header/WireGuardLayer.h

```diff
@@ -0,0 +1,596 @@
+		{}
+
+		/**
+		 * Converts the WireGuard layer to a string representation.
+		 *
+		 * @return String representation of the WireGuard layer
+		 */
+		std::string toString() const override;
+
+		/**
+		 * Returns the OSI model layer that this protocol belongs to.
+		 *
+		 * @return OSI model layer corresponding to the Network layer
+		 */
+		OsiModelLayer getOsiModelLayer() const override
+		{
+			return OsiModelNetworkLayer;
+		}
+
+		/**
+		 * Get the WireGuard message type in the form of a WireGuardMessageType enum.
+		 *
+		 * @return The message type as a WireGuardMessageType enum value.
+		 */
+		virtual WireGuardMessageType getWireGuardMessageType() const
+		{
+			return WireGuardMessageType::Unknown;
+		}
+	};
+
+	/**
+	 * @class WireGuardHandshakeInitiationLayer
+	 * Represents the Handshake Initiation message layer
+	 */
+	class WireGuardHandshakeInitiationLayer : public WireGuardLayer
+	{
+	private:
+#pragma pack(push, 1)
+		/**
+		 * @struct wg_handshake_initiation
+		 * Represents the Handshake Initiation message structure
+		 */
+		typedef struct wg_handshake_initiation : wg_common_header
+		{
+			/** Sender index */
+			uint32_t senderIndex;
+			/** Initiator's ephemeral public key */
+			uint8_t initiatorEphemeral[32];
+			/** Encrypted initiator's static key */
+			uint8_t encryptedInitiatorStatic[48];
+			/** Encrypted timestamp */
+			uint8_t encryptedTimestamp[28];
+			/** MAC1 field */
+			uint8_t mac1[16];
+			/** MAC2 field */
+			uint8_t mac2[16];
+		} wg_handshake_initiation;
+#pragma pack(pop)
+	public:
+		/**
+		 * A constructor that creates the layer from an existing packet raw data
+		 * @param[in] data A pointer to the raw data
+		 * @param[in] dataLen Size of the data in bytes
+		 * @param[in] prevLayer A pointer to the previous layer
+		 * @param[in] packet A pointer to the Packet instance where layer will be stored in
+		 */
+		WireGuardHandshakeInitiationLayer(uint8_t* data, size_t dataLen, Layer* prevLayer, Packet* packet)
+		    : WireGuardLayer(data, dataLen, prevLayer, packet)
+		{}
+
+		/**
+		 * A constructor that creates a new Handshake Initiation message
+		 * @param[in] senderIndex The sender's index
+		 * @param[in] initiatorEphemeral The initiator's ephemeral public key
+		 * @param[in] encryptedInitiatorStatic The encrypted initiator's static key
+		 * @param[in] encryptedTimestamp The encrypted timestamp
+		 * @param[in] mac1 The MAC1 field
+		 * @param[in] mac2 The MAC2 field
+		 */
+		WireGuardHandshakeInitiationLayer(uint32_t senderIndex, const uint8_t initiatorEphemeral[32],
+		                                  const uint8_t encryptedInitiatorStatic[48],
+		                                  const uint8_t encryptedTimestamp[28], const uint8_t mac1[16],
+		                                  const uint8_t mac2[16]);
+
+		/**
+		 * Get the sender index from the Handshake Initiation message.
+		 *
+		 * @return The sender index as a 32-bit integer.
+		 */
+		uint32_t getSenderIndex() const;
```

Comment:
We have getter methods for all fields (in this class and the other classes), but we don't have setter methods. Do we want to add setter methods also?

For example:
```cpp
void setSenderIndex(uint32_t senderIndex);
void setInitiatorEphemeral(std::array<uint8_t, 32> initEphemeral);
...
void setReceiverIndex(uint32_t receiverIndex);
...

```

Reply:
yes sure 👍